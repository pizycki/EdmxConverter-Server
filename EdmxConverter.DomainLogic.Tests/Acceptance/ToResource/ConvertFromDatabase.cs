using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using EdmxConverter.DomainLogic.Service;
using EdmxConverter.DomainLogic.Tests.Properties;
using Xunit;
using static LanguageExt.Prelude;

namespace EdmxConverter.DomainLogic.Tests.Acceptance.ToResource
{
    public class ConvertFromDatabase
    {
        [Fact]
        [UseReporter(typeof(DiffReporter))]
        [UseApprovalSubdirectory("Approved")]
        public void convert_selected_varbinary_from_MigrationHistory() =>
            Some(Resources.SampleDatabaseEdmx)
                .Map(sample => new Hex(sample))
                .Map(sample => new DatabaseEdmx(sample))
                .Bind(ConvertToResource.FromDatabaseModel)
                .Match(res => res.ToString(), () => None.ToString())
                .Verify();
    }
}