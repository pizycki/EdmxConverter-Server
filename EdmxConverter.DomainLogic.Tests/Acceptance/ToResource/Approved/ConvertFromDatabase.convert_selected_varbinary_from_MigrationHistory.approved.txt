H4sIAAAAAAAEAM1X227bOBB9X2D/geDTFkjFJH3ZBlKL1ImLoHVSVGnfaWnsEMuLyktgf1sf+kn7Czu6y1Ict+miKAIEIjVzZubwzFD+9+u3+PVGSXIP1gmjE3oSHVMCOjO50OuEBr96/jd9/erPP+LLXG3I59buRWmHntol9M774owxl92B4i5SIrPGmZWPMqMYzw07PT5+yU5OGCAERSxC4o9Be6GgWuByZnQGhQ9cLkwO0jX7+CatUMk1V+AKnkFCU64KCRfc8yV3EDXLJUJ42HhKzqXgmFUKckUJ19p47jHns08OUm+NXqcFbnB5uy0A7VZcOmhqOevNv7es49OyLNY7tlBZcN6oHwQ8edHwxMbuT2Kbdjwik5fIuN+WVVdsJnRuDCXjOGczaUubQzxHFZwAFyHMEdk1PuqEgnoq/47ILEgfLCQagrdcHpEPYSlF9g62t+Yf0IkOUg7TxYTx3c4Gbn2wpgDrtx9h1RRxlVPCdv3Y2LFzG/jUNb4NAp+vMTZfSujEwB51L/+3AKgobBRKFnzzHvTa3yUUHymZiw3k7U6D+kkL7Ct08jaMgsSsP53pmZWUc6HBNglMJD85RNR6c46uCbdbUY2bgu+l4Cjpc6jbJ6oU8lCeXUZ9o7K6U9uOZntaOl7wokDKBi3e7JC07u/Z8/THxa5qDJa5BzTfZdtF8sbyNYzeYmjMdC6s862QKZnlamI25n8Pt220AcVjPfeMt8bl884p7+u/8mxGcD2XcyxPgfZVpdBlVPf8xKsas1xy+0CfzIwMSu/rtce86zYZ+tc7U4SYjdIe08QmPI06fsz5Y5Idm3TRO+mOJBo3cjl8NU30U5tQgtTci7zSztZ5UFFpEKVf5EwKrLc3WHAtVuB8PRQp3ginoxvt97ldmHO5PHzF/PKhHrT4EgDZxGxWAuxPDXh9z212x+1fim+eDZF+dohPZ83BEb1vQtcqS2i+xMVtnV490Z84u6eCj9nwiy2+ACfWPUT5/aYhK5XUg7Y2V3plWoKxnGFGrcmI/wV4niMr5xYPkGceX2fgXHXRfuYyoMmlWkJ+pW+CL4I/dw7UUm6H9cbs8fjVBbWbc3xTlCv3f5SAaQosAW70myBk3uU9n+pvH0QpkLeA+1WH4ocGwq23HdK10d8J1NB3AQXosiluAW8SBHM3OuX38JTc8PviPax5tm3n1n6QwwexS3t8IfjacuUajN6//BXCyp8hr/4DhXQ0D7gMAAA=